### ISSUE TITLE
Some classes are not recognized as components or pipes in NgModule after upgrade to Angular 9

### ISSUE THREAD

**Issue by boyanio** [angular/angular#36199]
# 🐞 bug report

### Is this a regression?

It used to work fine with Angular 8

### Description

Once we upgraded to Angular 9, we got this problem. We use Nx to organize the project, we I do not believe it has anything to do with it. We have one publishable Angular library and one Angular application. The library export a NgModule with a lot of components & pipes. `npm start` works as expected. However, building in production fails.

I do `ng build my-lib --prod`, which creates a dist version in `dist/libs/my-lib` by ng-packagr. For the production build of the application, we add a TypeScript path `@work/my-lib` => `dist/libs/my-lib`, i.e. we use the created output by ng-packagr. When I do `ng build my-app --prod`, then we get the following errors for __some__ of the components:

```
error NG6001: The class 'XXX' is listed in the declarations of the NgModule 'YYY', but is not a directive, a component, or a pipe. Either remove it from the NgModule's declarations, or add an appropriate Angular decorator.
```

The class is a very simple component with the appropriate attribute. We have probably 50 components and we get this error for only 7 of them. What makes it more weird is that the order of import in the module "solves" some of the errors. I.e., in my `lib.module.ts` we can change the order of such imports

```
import { MyComponent1 } from 'components/my-component1.component.ts';
import { MyComponent2 } from 'components/my-component2.component.ts';
```

and some of the errors disappear. I tried, of course, to solve it this way first, but I cannot "resolve" all 7 errors.

## 🔬 Minimal Reproduction

I tried to reproduce this in a smaller set-up, but to no avail - with fewer components, it works well. So I hope you could point me to something, where I could give more information.

## 🌍  Your Environment

**Angular Version:**
<pre><code>
@angular-devkit/architect          0.900.6
@angular-devkit/build-angular      0.901.0-next.4
@angular-devkit/build-ng-packagr   0.901.0-next.4
@angular-devkit/build-optimizer    0.901.0-next.4
@angular-devkit/build-webpack      0.901.0-next.4
@angular-devkit/core               9.0.6
@angular-devkit/schematics         9.0.6
@angular/cdk                       9.1.3
@angular/cli                       9.1.0-next.4
@angular/flex-layout               9.0.0-beta.29
@angular/material                  9.1.3
@angular/material-moment-adapter   9.1.3
@ngtools/webpack                   9.1.0-next.4
@schematics/angular                9.0.6
@schematics/update                 0.901.0-next.4
ng-packagr                         9.0.3
rxjs                               6.5.4
typescript                         3.7.5
webpack                            4.41.2
</code></pre>

**Comment by flash-me** [comment]
Hi @boyanio , 

may you check if every declared Component / Pipe / Directive is also 
- **exported** the class itself
- **re-exported** in the "public_api.ts"

cheers
flash :zap:

**Comment by boyanio** [comment]
Hey @flash-me Thank you for the suggestion. Indeed, some components were not re-exported in public_api.ts However, I got a relevant error for those and it was easy fixing it. It did not help my initial problem though.

After playing around for some days, trying to cut my project down to something reproducible, I managed to solve the issue. It turned out, my tsconfig.json was missing `"importHelpers": true` I have no idea why this was cauisng issues only with certain components, but adding it, solved everything
